### ISSUE TITLE
Questions wrt. code

### ISSUE THREAD

**Issue by pfistfl** [VIPER-GENEVA/FlowCyt-Classification-Benchmark#4]
Hi, Thank you for putting together this benchmark and providing baseline implementations — it’s a valuable contribution, especially because FlowCap is extremely hard to access now.

While working with the codebase, I noticed a few things that might be worth reviewing. I may be misreading the code, so please correct me if I’m wrong.

1. Early stopping on test set
It appears that early stopping is evaluated against the test set. This could lead to optimistic performance estimates, as the model selection effectively uses test data. This might be a leftover from tests I assume?
https://github.com/VIPER-GENEVA/FlowCyt-Classification-Benchmark/blob/8c7927520268d1b032d8adec676549218c45514a/inductive/dnn_main.py#L150

2. Learning rate scheduler mode
The ReduceLROnPlateau scheduler appears to be configured with mode='min', but is monitoring accuracy rather than loss. Since higher accuracy is better, this should likely be mode='max' to correctly detect plateaus.
https://github.com/VIPER-GENEVA/FlowCyt-Classification-Benchmark/blob/8c7927520268d1b032d8adec676549218c45514a/inductive/dnn_main.py#L133

3. Data ordering
The data appears to be ordered by class rather than shuffled. This is primarily a concern when using mini-batch or stochastic gradient descent rather than full-batch updates. If class sizes are imbalanced — and in this case, the largest class appears at the end — it could bias gradient updates within each epoch. One could add a small warning there.

Best,

**Comment by LorenzoBini4** [comment]
Hi, thanks for you comment!

1. You’re right: in `inductive/dnn_main.py`  the variable named `best_val_acc` is being updated via:

`best_val_acc = test_dnn(test_loader, model, device) * 100.0
`
that effectively turns the test set into a model-selection signal, which can inflate reported performance. This is unintended and a leftover from quick sanity checks during development. We'll switch this to use a proper validation loader (e.g., `val_loader`) for early stopping / checkpoint selection, and only run `test_loader` once at the end using the selected checkpoint.

2. Also correct. If the scheduler is stepping on accuracy, then `mode` should be `max`.
 
3.  We'll ensure the train `DataLoader` uses `shuffle=True` by default. Additionally, we'll add a brief warning in the code/README noting that if users disable shuffling (or use custom sampling), they should be aware of class ordering effects. 

Thanks again for helping us improving our benchmark, cytometry community will appreciate.
